src/NServiceBus.Persistence.AzureStorage/Subscriptions/AzureSubscriptionStorage.cs

```diff
@@ -142,5 +190,11 @@ public int GetHashCode(Subscriber obj)
                 return (obj.Endpoint ?? string.Empty).Length;
             }
         }
+
+        internal class CacheItem
+        {
+            public DateTime Stored;
+            public Task<IEnumerable<Subscriber>> Subscribers;
```

Comment:
Nope. What I meant is that behind a task there can be a state machine/variables captured in a closure. I'm asking if awaiting the result and then re-wrapping it in a new Task is a better/cheaper approach, that holds less objects.
☝️ this might an over-engineering though.